app/src/main/java/ru/practicum/android/diploma/db/data/db/VacancyDbConvertor.kt

```diff
@@ -0,0 +1,168 @@
+package ru.practicum.android.diploma.db.data.db
+
+import ru.practicum.android.diploma.db.data.db.entity.AreaEntity
+import ru.practicum.android.diploma.db.data.db.entity.EmployerEntity
+import ru.practicum.android.diploma.db.data.db.entity.EmployerJoins
+import ru.practicum.android.diploma.db.data.db.entity.JobInfoEntity
+import ru.practicum.android.diploma.db.data.db.entity.JobInfoJoins
+import ru.practicum.android.diploma.db.data.db.entity.LogosEntity
+import ru.practicum.android.diploma.db.data.db.entity.SalaryEntity
+import ru.practicum.android.diploma.db.data.db.entity.SkillsEntity
+import ru.practicum.android.diploma.db.data.db.entity.VacancyEntity
+import ru.practicum.android.diploma.db.data.db.entity.VacancyJoins
+import ru.practicum.android.diploma.details.domain.model.AreaDetails
+import ru.practicum.android.diploma.details.domain.model.EmployerInfo
+import ru.practicum.android.diploma.details.domain.model.JobInfo
+import ru.practicum.android.diploma.details.domain.model.LogoUrlsDetails
+import ru.practicum.android.diploma.details.domain.model.SalaryDetails
+import ru.practicum.android.diploma.details.domain.model.VacancyDetails
+import java.text.SimpleDateFormat
+import java.util.Date
+
+class VacancyDbConvertor {
+
+    fun mapVacancy(vacancy: VacancyJoins): VacancyDetails {
+        return VacancyDetails(
+            id = vacancy.vacancy.id.toString(),
+            name = vacancy.vacancy.name,
+            description = vacancy.vacancy.description,
+            employerInfo = mapEmployer(vacancy.employerRow),
+            jobInfo = mapJobDetails(vacancy.jobInfoRow),
+            vacancyUrl = vacancy.vacancy.vacancyUrl
+        )
+    }
```

Comment:
Все подобные маперы можно писать вот так

```suggestion
    fun mapVacancy(vacancy: VacancyJoins): VacancyDetails =
        VacancyDetails(
            id = vacancy.vacancy.id.toString(),
            name = vacancy.vacancy.name,
            description = vacancy.vacancy.description,
            employerInfo = mapEmployer(vacancy.employerRow),
            jobInfo = mapJobDetails(vacancy.jobInfoRow),
            vacancyUrl = vacancy.vacancy.vacancyUrl
        )
```